byzcoin/bcadmin/main.go

```diff
@@ -477,20 +493,24 @@ func link(c *cli.Context) error {
 		}
 		ad := &darc.Darc{}
 		adPub := cothority.Suite.Point()
-		if adIDStr := c.String("darc"); len(adIDStr) == 69 {
-			adID, err := hex.DecodeString(adIDStr[5:])
+		// Accept both plain-darcs, as well as "darc:...." darcs
+		if adIDStr := c.String("admindarc"); len(adIDStr) == 64 || len(adIDStr) == 69 {
+			if len(adIDStr) == 69 {
+				adIDStr = adIDStr[5:]
+			}
+			adID, err := hex.DecodeString(adIDStr)
 			if err != nil {
 				return errors.New("couldn't parse given admin: " + err.Error())
 			}
-			var adPubBuf []byte
-			if pubStr := c.String("pub"); len(pubStr) == 72 {
-				adPubBuf, err = hex.DecodeString(pubStr[8:])
-				if len(adPubBuf) != 32 || err != nil {
-					return errors.New("please give valid admin public key in hex: " + err.Error())
-				}
-			} else {
-				return errors.New("please give a key in the following format: ed25519:public_key_in_hex")
+			adPubStr := c.String("adminpub")
+			// Accept both plain-public keys and "ed25519:..." keys
+			if (len(adPubStr) != 64 && len(adPubStr) != 72) || err != nil {
```

Comment:
Am really unconvinced on this length-based hacking around. If we are going to hack around on the input, strings.HasPrefix is much more self-documenting.